File: MiniPayPlatformFrontend/src/app/dto/transactionRequestDto.ts
export class TransactionRequestDto {
  paymentProviderId: number;
  amount: number;
  description: string;
  referenceId: string;

  constructor(
    paymentProviderId: number,
    amount: number,
    description: string,
    referenceId: string
  ) {
    this.paymentProviderId = paymentProviderId;
    this.amount = amount;
    this.description = description;
    this.referenceId = referenceId;
  }
}

File: MiniPayPlatformBackend/MiniPay.Application/DTOs/TransactionRequestDto.cs
namespace MiniPay.Application.DTOs {
	/*
	 * @brief This class represents a request to initiate a transaction.
	 */
	public class TransactionRequestDto {
		public required int PaymentProviderId { get; set; }
		public required decimal Amount { get; set; }
		public required string Description { get; set; }
		public required string ReferenceId { get; set; }
	}
}
